### PR TITLE
Voting edge case

### LINKED ISSUE #14: Setup tests and address voting edge cases.
From a chat with Dominik

> Hey Roberto, i have a question about the winning proposals. Your code filters winners with: 
> `yesVotes > noVotes`
> And losers with:
> `yesVotes < noVotes`
> What is the status of proposals that have?
> `yesVotes === noVotes`
> 
> And what happens when they receive no votes at all?
> Maybe i didn't see if at the daoBot code. 
> 
> And one more question, when i import the round data i receive very long floats for the ocean amounts in rounds 9 and 10, round 8 truncates after 2 decimals. Are you using a different algorithm after round 8 or is this some unforeseen behavior of airtable? Hope you have a great day 🙂
> 
> RA | Ocean — 10/11/2021
> Hmm... I should probably have put losers as yesVotes <= noVotes.
> You need above 50%
> 
> hmm... weird... I believe the column is set to be rounded to 2 decimals
> hmm, the column is set to be an integer...
> when i download the csv, it's an integer... I'll need to debug to see what's going on.
> 
> Maka — Yesterday at 3:32 PM
> maybe the problem is on my side, but its strange that the import for round 8 is working as intended 

#### DOD - Airtable Decimals
- [ ] Look into Airtable "Ocean Received" column, and identify the problem where we're seeing decimals instead of integers.
- [ ] Update script to preformat value before sending to Airtable, or an appropriate solution that addresses this problem. Round down to closest single digit seems safe.

#### DOD  - Vote Winners
- [x] Create unit tests & validate how vote count would be impacted, under each condition listed above in the ticket.
`YesVotes === NoVotes`.
- [x] Propose rules for winner/losers based on voter results as described by the ticket, fix script, and make sure the all tests are passing.

### PR DESCRIPTION
**Pull request by claudiaHash**
Fixes #14 .

Changes proposed in this PR:

- Voting edge case `YesVotes === NoVotes` addressed
- fix test impacted by the above change

### CODE REVIEW

File: src/test/airtable/test_airtable_round_winners.test.js [hunk 1]
```diff
@@ -47,19 +47,19 @@ beforeEach(async function () {
...
     },
     {
-      id: 'proposal_1',
+      id: 'proposal_2',
       fields: {
         'Project Name': 'Oblong Apple',
         'USD Requested': 20000,
         'Voted Yes': 1000,
-        'Voted No': 0
+        'Voted No': 1000
```
Comment (idiom-bytes) [inline src/test/airtable/test_airtable_round_winners.test.js:55]: Hi @claudiaHash, was this added to address Edge Case (2)?

> "What happens when a project get's 0 votes?"

If so, I would have expected my comment further below to address the notFundedResults.length to be 2
`should.equal(notFundedResults.length, 1)`

Thank you for updating the tests & validations!

File: src/test/airtable/test_airtable_round_winners.test.js [hunk 2]
```diff
@@ -336,8 +336,8 @@ describe('Calculating Winners', function () {
     const notFundedResults = await dumpResultsToGSheet(finalResults.notFunded)
 
     // Validate all winning, not funded, and downvoted proposals add up
-    should.equal(downvotedResults.length, 2)
-    should.equal(earmarkedResults.length, 3)
+    should.equal(downvotedResults.length, 3)
+    should.equal(earmarkedResults.length, 2)
     should.equal(generalResults.length, 6)
     should.equal(partiallyFundedResults.length, 1)
     should.equal(notFundedResults.length, 1)
```
Comment (idiom-bytes) [inline src/test/airtable/test_airtable_round_winners.test.js:343]: Can we double check that Use Case (2) be handled?

> When votes == 0

**Comment by idiom-bytes** [comment]
Hi Claudia, just read the discussions with @KinimodEtlon to review the DoD.
I believe the a couple things need to be resolved here.

#### DOD
- [ ] Use Case (1) - Downvoting should be (<=) and Winners should be (>)

I have reviewed your code and updates to the tests, and I believe this Use Case is addressed, great.

- [ ]Use Case (2) - What happens when a project receives 0 votes?

Maybe i didn't see if at the DAOBot code, but I believe this edge case is not explicitly tested again. Can we make sure that if a project receives 0 votes, that they do not receive funding?

Thank you!

File: src/test/airtable/test_airtable_round_winners.test.js (hunk 1 above)
Comment (claudiaHash) [reply src/test/airtable/test_airtable_round_winners.test.js:55]: Hi! thanks for checking and pointing that out, I'll make the needed check and push the changes

**Comment by idiom-bytes** [comment]
Hi Claudia, just looked at the code base & tests.
Thank you for providing coverage & confidence around the Y:0 N:0 outcome!

This looks ready to submit!

**Comment by claudiaHash** [comment]
> This looks ready to submit!

Thank you for taking time to review this!

**Review by KatunaNorbert — APPROVED** [review]
